index.js

```diff
@@ -113,8 +113,8 @@ function validateLabel(label, processingOption) {
   let error = false;
 
   if (normalize(label) !== label ||
-      (label[3] === "-" && label[4] === "-") ||
-      label[0] === "-" || label[label.length - 1] === "-" ||
+      (checkHyphens && label[3] === "-" && label[4] === "-") ||
```

Comment:
Very good point. Fixed.

Weirdly enough that doesn't reduce the failing tests. Seems like there're no test cases for this.